lib/ci/queue/redis/worker.rb

```diff
@@ -100,11 +105,24 @@ def should_requeue?(test)
           true
         end
 
-        def requeue(test)
+        REQUEUE = %{
+          local queue_key = KEYS[1]
+          local test = ARGV[1]
+          local offset = ARGV[2]
+
+          local pivot = redis.call('lrange', queue_key, -1 - offset, 0 - offset)[1]
+          if pivot then
+            redis.call('linsert', queue_key, 'BEFORE', pivot, test)
```

Comment:
Yes it is. With N being `size(queue) - offset`.

I wish there was an index based insert, but there isn't :/